### PR TITLE
24 initial development setup

### PR DESCRIPTION
**Pull request by Nikomus**
https://trello.com/c/RLaFuDpW/24-initial-development-setup-vvv-eee

- Adds a basic Next setup, a combination of the TypeScript and styled-components / ESLint, Prettier and Husky examples
- Adds License and other boilerplate

### CODE REVIEW

**Review by falconx — APPROVED** [review]
Looks good. I recommend checking out `emotion` as an alternative to `styled-components` https://github.com/jsjoeio/styled-components-vs-emotion. They share a very similar API, which makes transitioning pretty straightforward.

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
+  "parser": "@typescript-eslint/parser",
+  "plugins": ["@typescript-eslint"],
+  "extends": [
+    "eslint:recommended",
+    "plugin:react/recommended",
```
Comment (marxian) [inline .eslintrc.json:6]: I'd recommend considering:

`plugin:import/errors`, `plugin:react-hooks/recommended` and `plugin:jsx-a11y/recommended`

They've all improved my life at some point in the last couple of projects.

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    "browser": true,
+    "jest": true,
+    "node": true
+  },
+  "settings": {
+    "react": {
+      "version": "detect"
+    }
+  },
+  "rules": {
```
Comment (marxian) [inline .eslintrc.json:22]: If you do choose to use `jsx-a11y` you'll need:

```
'jsx-a11y/anchor-is-valid': [
      'error',
      {
        components: ['Link'],
        specialLink: ['hrefLeft', 'hrefRight'],
        aspects: ['invalidHref', 'preferButton'],
      },
    ],
```

File: .eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    "react/react-in-jsx-scope": 0,
+    "react/display-name": 0,
+    "react/prop-types": 0,
+    "@typescript-eslint/explicit-function-return-type": 0,
```
Comment (marxian) [inline .eslintrc.json:26]: I'm interested in some of these, they seem a little loose.

File: .prettierrc [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
+  "semi": false,
```
Comment (marxian) [inline .prettierrc:2]: :heart:

Comment (marxian) [reply .prettierrc:2]: It is time.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+      "pre-push": "yarn run type-check"
+    }
+  },
+  "lint-staged": {
+    "*.@(ts|tsx)": [
+      "yarn lint",
+      "yarn format"
+    ]
+  },
+  "dependencies": {
```
Comment (marxian) [inline package.json:25]: I'd like to suggest pinning your dependencies to strict versions, and keeping it that way with a `.yarnrc` rile with the content:

```
# Always install exact package versions
--add.exact true
```

**Review by marxian — APPROVED** [review]
Looks spiffy, apart from the sad lack of emojis in README.md, which I'd understood were essential these days?

As well as a few tweaks suggested in comments I'd also propose adding `.github/dependabot.yml` with cthe following content (seasoned to your taste)

``` 
# To get started with Dependabot version updates, you'll need to specify which
# package ecosystems to update and where the package manifests are located.
# Please see the documentation for all configuration options:
# https://help.github.com/github/administering-a-repository/configuration-options-for-dependency-updates

version: 2
updates:
  - package-ecosystem: 'npm' # See documentation for possible values
    directory: '/' # Location of package manifests
    schedule:
      interval: 'weekly'
      day: 'sunday'
    commit-message:
      prefix: 'fix:'
      prefix-development: 'chore:'
  - package-ecosystem: 'github-actions'
    directory: '/'
    schedule:
      interval: 'weekly'
      day: 'sunday'
    commit-message:
      prefix: 'fix:'
      prefix-development: 'chore:'
```

File: .prettierrc (hunk 4 above)
Comment (falconx) [reply .prettierrc:2]: :cry:

**Comment by Nikomus** [comment]
> Looks good. I recommend checking out `emotion` as an alternative to `styled-components` https://github.com/jsjoeio/styled-components-vs-emotion. They share a very similar API, which makes transitioning pretty straightforward.

Thank you, I have a separate card to add a proper styling system. I was thinking of sticking to styled-components but emotion does look good!

File: .eslintrc.json (hunk 1 above)
Comment (Nikomus) [reply .eslintrc.json:6]: I remember a slight run in with `jsx-a11y` settings in a previous project, but these are all good to include 🙂

**Comment by falconx** [comment]
> > Looks good. I recommend checking out `emotion` as an alternative to `styled-components` https://github.com/jsjoeio/styled-components-vs-emotion. They share a very similar API, which makes transitioning pretty straightforward.
> 
> Thank you, I have a separate card to add a proper styling system. I was thinking of sticking to styled-components but emotion does look good!

No worries. I originally made the switch to emotion because it's smaller, faster, and supports the same fundamental features. Happy to chat more or work with you if you do decide to change at any point.

> a proper styling system

I recommend looking at [styled-system](https://styled-system.com) if you haven't come across it before — it seems to now be standard for modern component libraries.

**Pull request merged**